### PR TITLE
feat(jspmify)

### PR DESCRIPTION
**Pull request by danthareja**
Adds full jspm support to it's own branch (closing #1). What do you think @Hendrixer @gdi2290?

**Notable changes**
* Bootstrap Angular manually in `app.js` now
* Removes Stylus support in favor of straight css (stylus imports are not yet supported by jspm)
* Updates karma tests to work with JSPM through `karma-jspm` plugin.
('angular' and 'angular-mocks' must now be imported by each test module [see here for deets](https://github.com/Workiva/karma-jspm/issues/23))
* Removes webpack build step from gulp, we only need to serve `System.js` and `jspm_packages` now with client.
* Updates generator in gulp to follow new import formats for components and specs
* Freezes jspm version locally to avoid systemJS 0.17 breaking changes.
*Updates installation/usage in README and adds new section that compares this approach to webpack

TODO: add jspm production build step to gulp. it's still a solid dev workflow right now.

### CODE REVIEW

**Comment by danthareja** [comment]
I've tested it in two different environments but still might be missing something. If you want to test it before merging, clone it down from [my branch](https://github.com/danthareja/NG6-starter/tree/jspm) and follow the updated instructions.

**Comment by gdi2290** [comment]
@Hendrixer could you review

File: package.json [hunk 1]
```diff
@@ -45,5 +13,36 @@
     "es6"
   ],
   "author": "AngularClass",
-  "license": "ISC"
+  "license": "ISC",
```
Comment (gdi2290) [inline package.json]: We haven't decided on MIT or Apache 2.0 (will probably go with MIT)

File: client/index.html [hunk 2]
```diff
@@ -1,12 +1,16 @@
 <!DOCTYPE html>
-<html lang="en" ng-app="app">
+<html lang="en">
 <head>
-	<meta charset="UTF-8">
-	<title>angular-webpack-starter</title>
-	<meta name="viewport" content="width=device-width, initial-scale=1">
+  <meta charset="UTF-8">
+  <title>angular-jspm-starter</title>
```
Comment (gdi2290) [inline client/index.html]: this change should be in master then pull from there

**Comment by gdi2290** [comment]
if we're making branches for each build process then master should be either a shell without one or just a readme that has links to either `webpack`, `jspm`, or `browserify` branch. If we're smart we can allow the workflow to be

```
# pseudo
git clone angular-class/NG6-starter .
git pull angular-class/NG6-starter#webpack
git pull angular-class/NG6-starter#styl
```
that might be possible

**Comment by danthareja** [comment]
latest commit tackles issue #7

**Comment by gdi2290** [comment]
@danthareja awesome thanks

**Comment by gdi2290** [comment]
@danthareja could you add a link to this branch on master's readme

**Comment by danthareja** [comment]
it shall be done!

**Pull request merged**